vault/provider.go

```diff
@@ -93,88 +113,342 @@ func Provider() terraform.ResourceProvider {
 				Description: "The namespace to use. Available only for Vault Enterprise",
 			},
 		},
+		ConfigureFunc:  providerConfigure,
+		DataSourcesMap: dataSourcesMap,
+		ResourcesMap:   resourcesMap,
+	}
+}
 
-		ConfigureFunc: providerConfigure,
+// Description is essentially a DataSource or Resource with some additional metadata
+// that helps with maintaining the Terraform Vault Provider.
+type Description struct {
+	// PathInventory is used for taking an inventory of the supported endpoints in the
+	// Terraform Vault Provider and comparing them to the endpoints noted as available in
+	// Vault's OpenAPI description. A list of Vault's endpoints can be obtained by,
+	// from Vault's home directory, running "$ ./scripts/gen_openapi.sh", and then by
+	// drilling into the paths with "$ cat openapi.json | jq ".paths" | jq 'keys[]'".
+	// Here's a short example of how paths and their path variables should be represented:
+	//		"/transit/keys/{name}/config"
+	//		"/transit/random"
+	//		"/transit/random/{urlbytes}"
+	//		"/transit/sign/{name}/{urlalgorithm}"
+	PathInventory []string
+
+	// Defaults to false, but should be marked true if a resource is enterprise only.
```

Comment:
I'm not sure how deep we go on lint checks, but you might consider the form:
```
// EnterpriseOnly defaults to...
```

Reference https://golang.org/doc/effective_go.html#commentary

Reply:
:+1: Will do.